test/system/src/test/java/io/pravega/test/system/KvsWithFailOverTest.java

```diff
@@ -0,0 +1,140 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.test.system;
+
+import io.pravega.client.ClientConfig;
+import io.pravega.client.KeyValueTableFactory;
+import io.pravega.client.admin.StreamManager;
+import io.pravega.client.admin.impl.StreamManagerImpl;
+import io.pravega.client.connection.impl.ConnectionPool;
+import io.pravega.client.connection.impl.ConnectionPoolImpl;
+import io.pravega.client.control.impl.ControllerImpl;
+import io.pravega.client.control.impl.ControllerImplConfig;
+import io.pravega.client.tables.impl.KeyValueTableFactoryImpl;
+import io.pravega.client.connection.impl.ConnectionFactory;
+import io.pravega.client.connection.impl.SocketConnectionFactoryImpl;
+import io.pravega.common.concurrent.ExecutorServiceHelpers;
+import io.pravega.common.concurrent.Futures;
+import io.pravega.common.hash.RandomFactory;
+import io.pravega.test.system.framework.Environment;
+import io.pravega.test.system.framework.SystemTestRunner;
+import io.pravega.test.system.framework.Utils;
+import io.pravega.test.system.framework.services.Service;
+import lombok.extern.slf4j.Slf4j;
+import mesosphere.marathon.client.MarathonException;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.Timeout;
+import org.junit.runner.RunWith;
+
+import java.net.URI;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+
+import static org.junit.Assert.assertTrue;
+
+@Slf4j
+@RunWith(SystemTestRunner.class)
+public class KvsWithFailOverTest extends AbstractFailoverTests {
+    static final String SCOPE_NAME = "kvsScope" + RandomFactory.create().nextInt(Integer.MAX_VALUE);
+    static final String KVT_NAME = "testKvt";
+    static final String KEY_FAMILY = "testKeyFamily";
+    private static final int NUM_KVPWRITE = 5;
+    private static final int NUM_KVPREAD = 5;
+    @Rule
+    //The execution time for @Before + @After + @Test methods should be less than 25 mins. Else the test will timeout.
+    public Timeout globalTimeout = Timeout.seconds(25 * 60);
```

Comment:
25 minutes sounds really long. Is this test really that slow?

Reply:
Update to 15 mins